plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/guru/ImplementNetworkVspCommand.java

```diff
@@ -112,13 +147,212 @@ public boolean isDefaultEgressPolicy() {
+            return false;
+        if (_networkCidr != null ? !_networkCidr.equals(that._networkCidr) : that._networkCidr != null) return false;
+        if (_networkDomainName != null ? !_networkDomainName.equals(that._networkDomainName) : that._networkDomainName != null)
+            return false;
+        if (_networkDomainPath != null ? !_networkDomainPath.equals(that._networkDomainPath) : that._networkDomainPath != null)
+            return false;
+        if (_networkDomainUuid != null ? !_networkDomainUuid.equals(that._networkDomainUuid) : that._networkDomainUuid != null)
+            return false;
+        if (_networkGateway != null ? !_networkGateway.equals(that._networkGateway) : that._networkGateway != null)
+            return false;
+        if (_networkName != null ? !_networkName.equals(that._networkName) : that._networkName != null) return false;
+        if (_networkUuid != null ? !_networkUuid.equals(that._networkUuid) : that._networkUuid != null) return false;
+        if (_vpcName != null ? !_vpcName.equals(that._vpcName) : that._vpcName != null) return false;
+        if (_vpcUuid != null ? !_vpcUuid.equals(that._vpcUuid) : that._vpcUuid != null) return false;
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = _networkDomainName != null ? _networkDomainName.hashCode() : 0;
+        result = 31 * result + (_networkDomainPath != null ? _networkDomainPath.hashCode() : 0);
+        result = 31 * result + (_networkDomainUuid != null ? _networkDomainUuid.hashCode() : 0);
+        result = 31 * result + (_networkAccountName != null ? _networkAccountName.hashCode() : 0);
+        result = 31 * result + (_networkAccountUuid != null ? _networkAccountUuid.hashCode() : 0);
+        result = 31 * result + (_networkName != null ? _networkName.hashCode() : 0);
+        result = 31 * result + (_networkCidr != null ? _networkCidr.hashCode() : 0);
+        result = 31 * result + (_networkGateway != null ? _networkGateway.hashCode() : 0);
+        result = 31 * result + (_networkAclId != null ? _networkAclId.hashCode() : 0);
+        result = 31 * result + (_dnsServers != null ? _dnsServers.hashCode() : 0);
+        result = 31 * result + (_gatewaySystemIds != null ? _gatewaySystemIds.hashCode() : 0);
+        result = 31 * result + (_networkUuid != null ? _networkUuid.hashCode() : 0);
+        result = 31 * result + (_isL3Network ? 1 : 0);
+        result = 31 * result + (_isVpc ? 1 : 0);
+        result = 31 * result + (_isSharedNetwork ? 1 : 0);
+        result = 31 * result + (_vpcName != null ? _vpcName.hashCode() : 0);
+        result = 31 * result + (_vpcUuid != null ? _vpcUuid.hashCode() : 0);
+        result = 31 * result + (_defaultEgressPolicy ? 1 : 0);
+        result = 31 * result + (_ipAddressRange != null ? _ipAddressRange.hashCode() : 0);
+        result = 31 * result + (_domainTemplateName != null ? _domainTemplateName.hashCode() : 0);
+        return result;
+    }
```

Comment:
Please implement ``toString()`` to provide debugging state information for debugging purposes.